daemon/daemon/daemon.go

```diff
@@ -18,9 +26,112 @@ func Start(ctx context.Context, server *rpc.Server) error {
 		}
 	}()
 
-	// Block here until context is cancelled
-	<-ctx.Done()
-	log.Info("Shutting down 40swapd")
+	// monitor every 10 seconds
+	for {
+		select {
+		case <-ctx.Done():
+			log.Info("Shutting down 40swapd")
+
+			return nil
+		default:
+			monitor := &SwapMonitor{
+				repository: db,
+				swapClient: swaps,
+			}
+			monitor.MonitorSwaps(ctx)
+
+			time.Sleep(MONITORING_INTERVAL_SECONDS * time.Second)
+		}
+	}
+}
+
+type SwapMonitor struct {
+	repository interface {
+		database.SwapInRepository
+	}
+	swapClient swaps.ClientInterface
+}
+
+func (m *SwapMonitor) MonitorSwaps(ctx context.Context) {
+	swapIns, err := m.repository.GetPendingSwapIns()
```

Comment:
Will this func monitor also swap outs?

Reply:
that is the idea. @markettes would need to add `m.repository.GetPendingSwapOuts()`